### PR TITLE
1378 dropdown event

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
- **Issue-1**: Drop down menu list width should be equal to dropdown
- **Issue-2:** Unable to select dropdown with keyboard
- **Issue-3:** Dropdown should clear when 'esc' key hit's single time

**Issue-1:**
image
image

**Review by Radha-Alberta — DISMISSED** [review]

**Comment by Radha-Alberta** [comment]
- filterable dropdown seems to be case sensitive, it should not
**Steps**:
1. Open filterable dropdown.
2. type 'r'
3. menu list should display 'Red'

image
image

**Comment by Radha-Alberta** [comment]
Error in Console

image

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 1]
```diff
@@ -296,23 +247,61 @@
...
     if (_disabled) return;
     value = option.value;
-    _selectedOption = option;
     if (!_native) {
       closeMenu();
-      _inputValue = option.label;
+      _inputElValue = option.label;
       _inputEl.focus();
     }
+    dispatchValue(value);
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/web-components/src/components/dropdown/Dropdown.svelte:262]: I don't think it will matter

**Comment by Radha-Alberta** [comment]
- **Issue-1:** pressing Tab should go to 'x' icon.
- **Issue-2:** Steps:
1. open dropdown menulist
2. select 'red' option
3. press 'Tab'
4. open the dropdown again, user can only see 1 option

**Comment by Radha-Alberta** [comment]
- Unable to add space when focus active on 'x' icon
**Steps**
1. focus active on 'x' icon.
2. user press space.

**Expected**: space should be added in dropdown input & menu list suggests 'No matches found'

File: libs/web-components/src/components/date-picker/DatePicker.svelte [hunk 2]
```diff
@@ -70,34 +70,44 @@
...
-    if (!_date) return;
-    
     if (["Space", "Enter"].includes(e.key)) {
       showCalendar();
       return;
     }
 
     switch (e.key) {
       case "ArrowLeft":
+        _date ||= addDays(new Date(), 1);
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/web-components/src/components/date-picker/DatePicker.svelte:80]: As mentioned in our conversation, to allow the date picker to start with the current date, the date is initialized with a date of one "unit" opposite of what the key that was pressed does. As you mentioned I will add a comment in the code to make sure this still makes sense a couple weeks from now :)

**Pull request merged**